### PR TITLE
Changing mapping addresses for pmem datapools

### PR DESCRIPTION
**Pull request by kasiawasiuta**
Some of tests are passing but they should not, due to changing address of datapool mapping:

- test_refcount

This change is [Reviewable](https://reviewable.io/reviews/pmem/pelikan/59)

### CODE REVIEW

**Comment by kasiawasiuta** [comment]
@pmem-bot-ae 
test this please

File: src/storage/slab/slab.c [hunk 1]
```diff
@@ -318,13 +319,22 @@ _slab_slabclass_teardown(void)
 static rstatus_i
 _slab_heapinfo_setup(void)
 {
+    void* rearrange_pool_mapping_addr = NULL;
     heapinfo.nslab = 0;
     heapinfo.max_nslab = slab_mem / slab_size;
 
     heapinfo.base = NULL;
     if (prealloc) {
+        if (pool_slab != NULL && slab_datapool != NULL) {
```
Comment (pbalcer) [inline src/storage/slab/slab.c]: You can't assume that if a global pointer is not NULL that it's valid. In this particular case I'm guessing that what you are doing is use-after-free.

**Review by pbalcer — COMMENTED** [review]
Is this so that you can force the datapool to pick a different address?
You should do this in the test instead. Either by hooking to datapool_open() to intercept the address or parsing the maps table after open to look for the pool mapping.

**Review by kasiawasiuta — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/pelikan/59)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @pbalcer)

___
*[src/storage/slab/slab.c, line 328 at r1](https://reviewable.io/reviews/pmem/pelikan/59#-LkTC3Yf5u4IpIcVECs1-r1-328:-Ll_bm4kA-8djVb-p0rX:bb2riff) ([raw file](https://github.com/pmem/pelikan/blob/1c4a9e352360148de5faec6b29839106691a16ec/src/storage/slab/slab.c#L328)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

You can't assume that if a global pointer is not NULL that it's valid. In this particular case I'm guessing that what you are doing is use-after-free.
</blockquote></details>

Yes thanks, I changed the implementation according to your suggestion

File: src/storage/slab/slab.c (hunk 1 above)
Comment (kasiawasiuta) [reply src/storage/slab/slab.c]: Yes thanks, I changed the implementation according to your suggestion

**Review by michalbiesek — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/pmem/pelikan/59#-:-Lnvt3ytDjcd4-lxKgjm:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @pbalcer)

**Pull request merged**